xrpl/models/ledger_objects/ledger_object.py

```diff
@@ -0,0 +1,171 @@
+"""A model for Ledger Objects"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Type, TypeVar
+
+from xrpl.models.base_model import BaseModel
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.ledger_objects.ledger_entry_type import LedgerEntryType
+from xrpl.models.required import REQUIRED
+
+# TODO: REMOVE if optional is needed
+# from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+L = TypeVar("L", bound="LedgerObject")
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class LedgerObject(BaseModel):
+    """The base model for a Ledger Object."""
+
+    index: Optional[str] = None
+    """
+    The unique ID for this ledger entry; either `index` or `LedgerIndex` is used. In
+    JSON, this field is represented with different names depending on the context and
+    API method. (Note, even though this is specified as "optional" in the code, every
+    ledger entry should have one unless it's legacy data from very early in the XRP
+    Ledger's history.)
+    """
+
+    ledger_index: Optional[str] = None
+    """
+    The unique ID for this ledger entry; either `index` or `LedgerIndex` is used. In
+    JSON, this field is represented with different names depending on the context and
+    API method. (Note, even though this is specified as "optional" in the code, every
+    ledger entry should have one unless it's legacy data from very early in the XRP
+    Ledger's history.)
+    """
+
+    ledger_entry_type: LedgerEntryType = REQUIRED  # type: ignore
+    """
+    The type of ledger entry. Valid ledger entry types include AccountRoot, Offer,
+    RippleState, and others. This field is required.
+    """
+
+    flags: int = REQUIRED  # type: ignore
```

Comment:
```suggestion
    flags: Union[FlagInterface, int, List[int]] = 0
```
This matches how the `Transaction` base class does it, and also allows support for the flag interfaces created.